File: src/types/claude.ts
export interface ClaudeAccount {
  id: string;
  email: string;
  name?: string | null;
  tags?: string[] | null;

  config_dir: string;
  login_mode: string;
  login_hint_email?: string | null;
  anthropic_base_url?: string | null;
  anthropic_auth_token?: string | null;
  disable_nonessential_traffic?: boolean;

  logged_in: boolean;
  auth_method?: string | null;
  api_provider?: string | null;
  org_id?: string | null;
  org_name?: string | null;
  subscription_type?: string | null;

  status_raw?: unknown;

  created_at: number;
  last_used: number;
  last_synced_at?: number | null;
}

export interface ClaudeUsage {
  inlineSuggestionsUsedPercent: number | null;
  chatMessagesUsedPercent: number | null;
  totalPercentUsed: number | null;
}

export function getClaudeAccountDisplayName(account: ClaudeAccount): string {
  const email = account.email?.trim();
  if (email) return email;
  const name = account.name?.trim();
  if (name) return name;
  const hint = account.login_hint_email?.trim();
  if (hint) return hint;
  const baseUrl = account.anthropic_base_url?.trim();
  if (baseUrl) return baseUrl;
  return account.id;
}

export function getClaudeLoginModeLabel(account: Pick<ClaudeAccount, 'login_mode'>): string {
  switch ((account.login_mode || '').trim().toLowerCase()) {
    case 'auth_token':
      return 'Auth Token';
    case 'console':
      return 'Console';
    case 'sso':
      return 'SSO';
    case 'email':
      return 'Email Hint';
    case 'claudeai':
    default:
      return 'Claude.ai';
  }
}

export function getClaudePlanBadge(account: ClaudeAccount): string {
  const subscription = (account.subscription_type || '').trim();
  if (subscription) return subscription.toUpperCase();
  if ((account.login_mode || '').trim().toLowerCase() === 'auth_token') {
    return account.logged_in ? 'TOKEN' : 'ENV';
  }
  return account.logged_in ? 'ACTIVE' : 'PROFILE';
}

export function getClaudePlanBadgeClass(account: ClaudeAccount): string {
  const subscription = (account.subscription_type || '').trim().toLowerCase();
  if (!subscription) return account.logged_in ? 'pro' : 'unknown';
  if (subscription.includes('enterprise') || subscription.includes('team')) return 'enterprise';
  if (subscription.includes('pro') || subscription.includes('max')) return 'pro';
  if (subscription.includes('free')) return 'free';
  return 'unknown';
}

export function getClaudeUsage(): ClaudeUsage {
  return {
    inlineSuggestionsUsedPercent: null,
    chatMessagesUsedPercent: null,
    totalPercentUsed: null,
  };
}

File: src-tauri/src/models/claude.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ClaudeAccount {
    pub id: String,
    pub email: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,

    pub config_dir: String,
    pub login_mode: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub login_hint_email: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub anthropic_base_url: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub anthropic_auth_token: Option<String>,
    #[serde(default)]
    pub disable_nonessential_traffic: bool,

    pub logged_in: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub auth_method: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub api_provider: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub org_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub org_name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub subscription_type: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub status_raw: Option<serde_json::Value>,

    pub created_at: i64,
    pub last_used: i64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_synced_at: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ClaudeAccountSummary {
    pub id: String,
    pub email: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    pub login_mode: String,
    pub logged_in: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub subscription_type: Option<String>,
    pub created_at: i64,
    pub last_used: i64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub last_synced_at: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ClaudeAccountIndex {
    pub version: String,
    pub accounts: Vec<ClaudeAccountSummary>,
}

impl ClaudeAccountIndex {
    pub fn new() -> Self {
        Self {
            version: "1.0".to_string(),
            accounts: Vec::new(),
        }
    }
}

impl Default for ClaudeAccountIndex {
    fn default() -> Self {
        Self::new()
    }
}

impl ClaudeAccount {
    pub fn summary(&self) -> ClaudeAccountSummary {
        ClaudeAccountSummary {
            id: self.id.clone(),
            email: self.email.clone(),
            name: self.name.clone(),
            tags: self.tags.clone(),
            login_mode: self.login_mode.clone(),
            logged_in: self.logged_in,
            subscription_type: self.subscription_type.clone(),
            created_at: self.created_at,
            last_used: self.last_used,
            last_synced_at: self.last_synced_at,
        }
    }
}
